src/main/java/org/opentripplanner/middleware/bugsnag/BugsnagDispatcher.java

```diff
@@ -0,0 +1,161 @@
+        getProjects();
+
+        EVENT_DATA_REQUEST_FILTER
+            = FileUtils.getFileContents("src/main/resources/org/opentripplanner/middleware/eventDataRequestFilter.json");
+
+        // replace placeholder with reporting window number of days
+        EVENT_DATA_REQUEST_FILTER
+            = EVENT_DATA_REQUEST_FILTER.replace("$BUGSNAG_REPORTING_WINDOW_IN_DAYS$",
+            String.valueOf(BUGSNAG_REPORTING_WINDOW_IN_DAYS));
+    }
+
+    /**
+     * Make the initial event data request to Bugsnag. This triggers an asynchronous job to prepare the data for one
+     * single download. The returned event data request can be used to check the status of the request
+     */
+    public static BugsnagEventRequest makeEventDataRequest() {
+
+        if (ORGANIZATION == null) {
+            return null;
+        }
+
+        URI uri = HttpUtils.buildUri(BUGSNAG_API_URL,
+            "/organizations/" + ORGANIZATION.id + "/event_data_requests",
+            null);
+
+        String response = HttpUtils.httpRequest(uri,
+            CONNECTION_TIMEOUT_IN_SECONDS,
+            HttpUtils.REQUEST_METHOD.POST,
+            BUGSNAG_HEADERS, EVENT_DATA_REQUEST_FILTER);
+
+        BugsnagEventRequest eventDataRequest = JsonUtils.getPOJOFromJSON(response, BugsnagEventRequest.class);
+
+        return eventDataRequest;
+    }
+
+    /**
+     * Get a previously created event data request. A status of 'complete' signals that the requested data is ready to
+     * be downloaded from the populated url parameter.
+     */
+    public static BugsnagEventRequest getEventDataRequest(BugsnagEventRequest originalEventDataRequest) {
+
+        if (ORGANIZATION == null) {
+            return null;
+        }
+
+        String endpoint = "/organizations/" + ORGANIZATION.id + "/event_data_requests/" +
+            originalEventDataRequest.eventDataRequestId;
+
+        URI uri = HttpUtils.buildUri(BUGSNAG_API_URL, endpoint, null);
+
+        String response = HttpUtils.httpRequest(uri,
+            CONNECTION_TIMEOUT_IN_SECONDS,
+            HttpUtils.REQUEST_METHOD.GET, BUGSNAG_HEADERS, null);
+
+        BugsnagEventRequest eventDataRequest = JsonUtils.getPOJOFromJSON(response, BugsnagEventRequest.class);
+
+        return eventDataRequest;
+    }
+
+
+    /**
+     * Get a single organisation matching the organization held in the property file
+     */
+    private static void getOrganization() {
+
+        URI uri = HttpUtils.buildUri(BUGSNAG_API_URL, "/user/organizations/", null);
+        String response = HttpUtils.httpRequest(uri,
+            CONNECTION_TIMEOUT_IN_SECONDS,
+            HttpUtils.REQUEST_METHOD.GET, BUGSNAG_HEADERS, null);
+        Organization[] organizations = JsonUtils.getPOJOFromJSON(response, Organization[].class);
+
+        for (Organization org : organizations) {
+            if (org.name.equalsIgnoreCase(BUGSNAG_ORGANIZATION)) {
+                ORGANIZATION = org;
+                break;
+            }
+        }
```

Comment:
Should this throw an error if the organization isn't found?

Reply:
Yes. Updated.